lib/features/login/presentation/cubit/login_state.dart

```diff
@@ -0,0 +1,38 @@
+part of 'login_cubit.dart';
+
+abstract class LoginState extends Equatable {
+  const LoginState();
+}
+
+class LoginUIState extends LoginState {
+  final bool loading;
+  final String? usernameValidation;
+  final String? passwordValidation;
+
+  const LoginUIState({
+    this.loading = false,
+    this.usernameValidation,
+    this.passwordValidation,
+  });
+
+  @override
+  List<Object?> get props => [
+        loading,
```

Comment:
loading need to be excluded here if you want to operate between two changes on user and password only